### PR TITLE
Williamhe/add oryx server command

### PR DESCRIPTION
**Pull request by william-msft**
- [ ] The purpose of this PR is explained in this message or in an issue. If an issue please include a reference as https://devdiv.visualstudio.com/DevDiv/_boards/board/t/Oryx/Stories/?workitem=1463216
This code branches from: https://github.com/microsoft/Oryx/pull/1097/files
- [ ] Tests are included and/or updated for code changes.
- [ ] Proper license headers are included in each file.

### CODE REVIEW

File: src/BuildServer/BuildServerResponse.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+
+namespace Microsoft.Oryx.BuildServer
+{
+    enum BuildState
```
Comment (qianz2) [inline src/BuildServer/BuildServerResponse.cs:10]: Is 'BuildState' being used or not? I couldn't find a place where it's used. If not, should we remove it

File: src/BuildServer/BuildServerRequests.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+namespace Microsoft.Oryx.BuildServer
+{
+    public class BuildServerRequests
+    {
+        [JsonProperty(PropertyName = "command")]
+        public string command { get; set; }
+
+        [JsonProperty(PropertyName = "id")]
+        public string id { get; set; }
```
Comment (qianz2) [inline src/BuildServer/BuildServerRequests.cs:16]: How about id -> buildRequestId

File: src/BuildServer/BuildServerRequests.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+        [JsonProperty(PropertyName = "status")]
+        public string? Status { get; set; }
+
+        [JsonProperty(PropertyName = "outputPath")]
+        public string OutputPath { get; set; }
```
Comment (qianz2) [inline src/BuildServer/BuildServerRequests.cs:22]: Do we still want 'outputPath' as a property if it's included in 'command'?

File: src/BuildServer/BuildServerRequests.cs (hunk 2 above)
Comment (william-msft) [reply src/BuildServer/BuildServerRequests.cs:16]: This file is actually not used. I'm removing this file. Thanks Stella.

File: src/BuildServer/BuildServerRequests.cs (hunk 3 above)
Comment (william-msft) [reply src/BuildServer/BuildServerRequests.cs:22]: This file is actually not used. I'm removing this file. Thanks Stella.

File: src/BuildServer/BuildServerResponse.cs (hunk 1 above)
Comment (william-msft) [reply src/BuildServer/BuildServerResponse.cs:10]: This file is actually not used. I'm removing this file. Thanks Stella.

File: README.md [hunk 4]
```diff
@@ -90,6 +90,35 @@ docker run --detach --rm \
...
+                --log-file [logPath]
+            ```
+1. Check build status with id `1`
+    1. ![Status](doc/status.png)
+1. Check server healthcheck
+    1. ![Health Check](doc/healthCheck.png)
+
 # Components
 
 Oryx consists of a build image, a collection of runtime images, a build script generator, and a collection of
```
Comment (qianz2) [inline README.md:124]: We can move below images into a doc/BuildServer folder.

File: src/BuildServer/Startup.cs [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using JsonFlatFileDataStore;
+using Microsoft.Oryx.BuildServer.Repositories;
+using Microsoft.Oryx.BuildServer.Services;
+using Microsoft.Oryx.BuildServer.Services.ArtifactBuilders;
+using System.IO;
+
+namespace Microsoft.Oryx.BuildServer
```
Comment (qianz2) [inline src/BuildServer/Startup.cs:19]: Is it possible to add tests for its functionality?

File: src/BuildServer/Services/ArtifactBuilders/Builder.cs [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    public class Builder : IArtifactBuilder
+    {
+        private readonly ILogger<Builder> _logger;
+
+        public Builder(ILogger<Builder> logger)
+        {
+            _logger = logger;
+        }
+
+        public bool Build(Build build)
```
Comment (qianz2) [inline src/BuildServer/Services/ArtifactBuilders/Builder.cs:43]: Can we add tests to verify the outputs from this method

File: src/BuildServer/ProcessAsTaskHelper.cs [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+using System.Diagnostics;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Oryx.BuildServer
+{
+    public static class ProcessAsTaskHelper
+    {
+        public static async Task<ProcessResults> RunProcessAsync(
```
Comment (qianz2) [inline src/BuildServer/ProcessAsTaskHelper.cs:18]: Is this helper class being used?

File: Oryx.sln [hunk 8]
```diff
@@ -96,7 +96,9 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Common", "src\Oryx.Common\C
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "BuildScriptGenerator.Common.Tests", "tests\BuildScriptGenerator.Common.Tests\BuildScriptGenerator.Common.Tests.csproj", "{3078BCDB-FC35-4826-82A1-9CE094446DE8}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "AutoUpdater", "build\tools\AutoUpdater\AutoUpdater.csproj", "{6C4DBA89-3D17-451C-9D89-71A70410D835}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "AutoUpdater", "build\tools\AutoUpdater\AutoUpdater.csproj", "{6C4DBA89-3D17-451C-9D89-71A70410D835}"
```
Comment (qianz2) [inline Oryx.sln:99]: Is AutoUpdater.csproj a project you're currently working on? If so, do we consider make it as a seperate PR or will add into this one afterwards?

File: README.md (hunk 4 above)
Comment (william-msft) [reply README.md:124]: Updated

File: src/BuildServer/ProcessAsTaskHelper.cs (hunk 7 above)
Comment (william-msft) [reply src/BuildServer/ProcessAsTaskHelper.cs]: No, it has been removed now.

File: Oryx.sln (hunk 8 above)
Comment (william-msft) [reply Oryx.sln]: No, it has been removed now.

**Review by qianz2 — APPROVED** [review]

**Pull request merged**